components/motor/gpio/loop.go

```diff
@@ -32,57 +29,73 @@ func (m *EncodedMotor) State(ctx context.Context) ([]float64, error) {
 func (m *EncodedMotor) updateControlBlock(ctx context.Context, setPoint, maxVel float64) error {
 	// Update the Trapezoidal Velocity Profile block with the given maxVel for velocity control
 	velConf := control.BlockConfig{
-		Name: "trapz",
+		Name: m.blockNames["trapz"],
 		Type: "trapezoidalVelocityProfile",
 		Attribute: rdkutils.AttributeMap{
 			"max_vel":    maxVel,
 			"max_acc":    30000.0,
 			"pos_window": 0.0,
 			"kpp_gain":   0.45,
 		},
-		DependsOn: []string{"set_point", "endpoint"},
+		DependsOn: []string{m.blockNames["constant"], m.blockNames["endpoint"]},
 	}
-	if err := m.loop.SetConfigAt(ctx, "trapz", velConf); err != nil {
+	if err := m.loop.SetConfigAt(ctx, m.blockNames["trapz"], velConf); err != nil {
 		return err
 	}
 
 	// Update the Constant block with the given setPoint for position control
 	posConf := control.BlockConfig{
-		Name: "set_point",
+		Name: m.blockNames["constant"],
 		Type: "constant",
 		Attribute: rdkutils.AttributeMap{
 			"constant_val": setPoint,
 		},
 		DependsOn: []string{},
 	}
-	if err := m.loop.SetConfigAt(ctx, "set_point", posConf); err != nil {
+	if err := m.loop.SetConfigAt(ctx, m.blockNames["constant"], posConf); err != nil {
 		return err
 	}
 	return nil
 }
 
 // validateControlConfig ensures the programmatically edited blocks are named correctly.
 func (m *EncodedMotor) validateControlConfig(ctx context.Context) error {
-	constBlock, err := m.loop.ConfigAt(ctx, "set_point")
+	m.blockNames = make(map[string]string)
+
+	// These three blocks are the only block names that are used by EncodedMotor
+	// verify constant block exits and store its name
+	constBlocks, err := m.loop.ConfigAtType(ctx, "constant")
+	m.logger.CDebugf(ctx, "const blocks = %v", constBlocks)
```

Comment:
Same question here about why bother to pass the context to `CDebugf` instead of merely using `Debugf`, and then the same suggestion to use `Debugw` instead.